### DISCUSSION TITLE
YAML metadata to trigger custom functions that populate missing values

### DISCUSSION THREAD

**Discussion by benbooth493** [cuelang/cue#1069] [Q&A]
Let’s say I have a yaml file with some values that can be empty.
Now say I have a program that reads the yaml and populates some of the values.
The values to populate is determined by a json schema document with `additionalProperties: true` and with some of the values having a custom property `mustPopulate: true`.

I am not using the json schema for validation (although I could of course), it’s really just to carry some additional metadata for the yaml file.

Is this pattern possible with Cue? If so, how would I do such a thing?

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/1069.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
